implementations/rust/ockam/ockam_executor/src/executor.rs

```diff
@@ -12,21 +12,31 @@ use ockam_core::compat::sync::Arc;
 use ockam_core::compat::task::Wake;
 
 /// Returns current executor.
-/// WARNING: TODO this is not thread-safe
+static INIT: AtomicBool = AtomicBool::new(false);
+static mut EXECUTOR: UnsafeCell<MaybeUninit<Executor>> = UnsafeCell::new(MaybeUninit::uninit());
+
 pub fn current() -> &'static Executor<'static> {
-    static INIT: AtomicBool = AtomicBool::new(false);
```

Comment:
BTW, whoever is going to review this; I got this code out to make it a singleton. Is that OK ?